instrumentation/google.golang.org/grpc/otelgrpc/config.go

```diff
@@ -203,45 +154,42 @@ func (m messageEventsProviderOption) apply(c *config) {
 //   - ReceivedEvents: Record the number of bytes read after every gRPC read operation.
 //   - SentEvents: Record the number of bytes written after every gRPC write operation.
 func WithMessageEvents(events ...Event) Option {
-	return messageEventsProviderOption{events: events}
-}
-
-type spanStartOption struct{ opts []trace.SpanStartOption }
-
-func (o spanStartOption) apply(c *config) {
-	c.SpanStartOptions = append(c.SpanStartOptions, o.opts...)
+	return optionFunc(func(c *config) {
+		for _, e := range events {
+			switch e {
+			case ReceivedEvents:
+				c.ReceivedEvent = true
+			case SentEvents:
+				c.SentEvent = true
+			}
+		}
+	})
 }
 
 // WithSpanOptions configures an additional set of
 // trace.SpanOptions, which are applied to each new span.
 //
 // Deprecated: It is only used by the deprecated interceptor, and is unused by [NewClientHandler] and [NewServerHandler].
 func WithSpanOptions(opts ...trace.SpanStartOption) Option {
-	return spanStartOption{opts}
-}
-
-type spanAttributesOption struct{ a []attribute.KeyValue }
-
-func (o spanAttributesOption) apply(c *config) {
-	if o.a != nil {
-		c.SpanAttributes = o.a
-	}
+	return optionFunc(func(c *config) {
+		c.SpanStartOptions = append(c.SpanStartOptions, opts...)
+	})
 }
 
 // WithSpanAttributes returns an Option to add custom attributes to the spans.
 func WithSpanAttributes(a ...attribute.KeyValue) Option {
-	return spanAttributesOption{a: a}
-}
-
-type metricAttributesOption struct{ a []attribute.KeyValue }
-
-func (o metricAttributesOption) apply(c *config) {
-	if o.a != nil {
-		c.MetricAttributes = o.a
-	}
+	return optionFunc(func(c *config) {
+		if a != nil {
+			c.SpanAttributes = a
+		}
+	})
 }
 
 // WithMetricAttributes returns an Option to add custom attributes to the metrics.
 func WithMetricAttributes(a ...attribute.KeyValue) Option {
-	return metricAttributesOption{a: a}
+	return optionFunc(func(c *config) {
+		if a != nil {
+			c.MetricAttributes = a
```

Comment:
For the reviewer's reference, to assess whether an adjustment is needed to: `c.MetricAttributes = append(c.MetricAttributes, a...)`

Reply:
Done